src/math/CurveFitting.cc

```diff
@@ -210,10 +189,8 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
                                                                                                         ndarray::makeVector(1))));
     LOGLS_DEBUG(_log, "P_D_A1_MeasureErrorsTemp = " << *P_D_A1_MeasureErrorsTemp);
 
-    for (size_t i = 0; i < I_A1_OrigPos.size(); ++i)
-      I_A1_OrigPos[i] = i;
     int I_NRejected_Old = 0;
-    std::vector<size_t> I_A1_Rejected_Old(D_A1_X_In.size());
+    std::vector<size_t> I_A1_Rejected_Old(0);
```

Comment:
Before using `I_A1_Rejected_Old` it is resized to size 0, and then filled with `std::vector::push_bak`. I thought it is easier to read this way: declaring it with size 0 and reserving the space for the size of `D_A1_X_In`. Testing that this behavior is correct would require to rip this function apart into lots of small functions which could be tested individually. As all tests are passing I assume that this new behavior is correct.